src/components/Calculator.js

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+
+const calcButtons = ['AC', '+/-', '%', '/', '7', '8', '9', 'X', '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];
+
+function Calculator() {
+  function CreateElement(text, index) {
+    let className = '';
+    if ((index + 1) % 4 === 0 || (index + 1) === 19) {
+      className = ' makeMe-orange';
+    } else if ({ text }.text === '0') {
+      className = ' calc-button-0';
+    }
+
+    return (
+
+      <button key={text} className={className} type="button">{text}</button>
+
+    );
+  }
+
+  return (
+    calcButtons.map(CreateElement)
+  );
+}
+
+export default Calculator;
```

Comment:
Hi @abdulhamiid i have a question. I think i already create a child. Because all my calc buttons are created inside of my App().
`function App() {
  return (
    <main className="calc-divs">
      <div className="row-total">
        <h2 className="calc-total">0</h2>
      </div>

      <div className="calc-grid">
        <Calculator />
      </div>
    </main>
  );
}`

All the elements of <Calculator /> are childs of App() am I alright?

Reply:
Hi @Zven94

Reply:
According to the [project requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_components.md) we are required to create at least a child component in the `Calculator.js`

![image](https://github.com/Zven94/math-magicians/assets/78251927/6d34e6ee-a74b-44d9-bc24-3c3cb8b3a79c)

Reply:
but it doesn't say **in calculator** as the others requeriments said....